packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -70,6 +77,7 @@ export type Actions =
   | ResultViewTypeChangedAction;
 
 export type State = {
+  pipeline: Document[];
```

Comment:
It was a bit cumbersome that results slice relies on the pipeline slice to get the pipeline to run: at the point when you press run results we generate pipeline once and it will not change until you go back to editor. I updated the slice to follow that UI logic

Reply:
sorry, i didn't understand what was cumbersome? I do like the change though.

Reply:
Was just a bit weird to me that we derive the data that doesn't change from another slice after you switch view every time you paginate or count documents. It was already cleaner now when pipelineBuilder is the source, but anyway made more sense to me to do this just once and keep it in state, I am glad it looks good to you